source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRShowCardTarget.mm

```diff
@@ -35,16 +36,103 @@ - (instancetype)initWithAdaptiveCard:(std::shared_ptr<AdaptiveCards::AdaptiveCar
+    }
+    ACRContentHoldingUIView *wrappingView = [[ACRContentHoldingUIView alloc]
+                                             initWithFrame:CGRectMake(0,0,
+                                                                      adcView.frame.size.width +
+                                                                      padding +
+                                                                      padding,
+                                                                      adcView.frame.size.height +
+                                                                      padding +
+                                                                      padding)];
+    [wrappingView addSubview:adcView];
+    NSString *horString = [[NSString alloc] initWithFormat:@"H:|-%u-[adcView]-%u-|",
+                           padding,
+                           padding];
+    NSString *verString = [[NSString alloc] initWithFormat:@"V:|-%u-[adcView]-%u-|",
+                           padding,
+                           padding];
+    NSDictionary *dictionary = NSDictionaryOfVariableBindings(wrappingView, adcView);
+    NSArray *horzConst = [NSLayoutConstraint constraintsWithVisualFormat:horString
+                                                                 options:0
+                                                                 metrics:nil
+                                                                   views:dictionary];
+    NSArray *vertConst = [NSLayoutConstraint constraintsWithVisualFormat:verString
+                                                                 options:0
+                                                                 metrics:nil
+                                                                   views:dictionary];
+    [wrappingView addConstraints:horzConst];
+    [wrappingView addConstraints:vertConst];
+    _adcView = wrappingView;
+
+    ContainerStyle style = (_config->adaptiveCard.allowCustomStyle)? _adaptiveCard->GetStyle() : _config->actions.showCard.style;
+
+    long num = 0;
+
+    if(style == ContainerStyle::None)
+    {
+        style = [_superview getStyle];
+    }
+
+    if(style == ContainerStyle::Emphasis)
+    {
+        num = std::stoul(_config->containerStyles.emphasisPalette.backgroundColor.substr(1), nullptr, 16);
+    }
+    else
+    {
+        num = std::stoul(_config->containerStyles.defaultPalette.backgroundColor.substr(1), nullptr, 16);
+    }
+
+    wrappingView.translatesAutoresizingMaskIntoConstraints = NO;
+    wrappingView.backgroundColor =
+    [UIColor colorWithRed:((num & 0x00FF0000) >> 16) / 255.0
+                    green:((num & 0x0000FF00) >>  8) / 255.0
+                     blue:((num & 0x000000FF)) / 255.0
+                    alpha:((num & 0xFF000000) >> 24) / 255.0];
+    [wrappingView setAlignmentForSubview:AdaptiveCards::HorizontalAlignment::Center];
+    [_superview addArrangedSubview:_adcView];
+}
+
+- (IBAction)showCard
+{
+    if(YES == _isItCurrentlyShown)
```

Comment:
>_isItCurrentlyShown [](start = 14, length = 19)

It looks like `_isItCurrentlyShown` is equivalent to `_adcView && !_adcView.hidden`. If that's right, it's better to remove the redundant field.